controllers/openstackdataplanenodeset_controller.go

```diff
@@ -417,6 +417,24 @@ func (r *OpenStackDataPlaneNodeSetReconciler) Reconcile(ctx context.Context, req
 		return ctrl.Result{}, nil
 	}
 
+	isDeploymentReady, isDeploymentRunning, isDeploymentFailed, err := checkDeployment(helper, instance)
+	if err != nil {
+		instance.Status.Conditions.MarkFalse(
+			condition.DeploymentReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityError,
+			condition.DeploymentReadyErrorMessage,
+			err.Error())
+		Log.Error(err, "Unable to get deployed OpenStackDataPlaneDeployments.")
+		return ctrl.Result{}, err
+	}
```

Comment:
Test case for this would be a good idea.

Reply:
it is already covered in the kuttl test:
https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/tests/kuttl/tests/dataplane-deploy-no-nodes-test/05-assert.yaml

Reply:
ack.